### PR TITLE
chore: add validators

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] Update password validator
- [x] Update author validator

### CODE REVIEW

File: cmd/server/router.go [hunk 1]
```diff
@@ -89,6 +92,10 @@ func publicHandler(r *gin.Engine, a App) {
 }
 
 func authenticatedHandler(r *gin.Engine, a App) {
+	if v, ok := binding.Validator.Engine().(*validator.Validate); ok {
+		v.RegisterValidation("passwordvalidator", view.PasswordValidator)
+		v.RegisterValidation("authorvalidator", view.AuthorValidator)
```
Comment (hieuphq) [inline cmd/server/router.go:97]:
```suggestion
		v.RegisterValidation("password", view.PasswordValidator)
		v.RegisterValidation("author", view.AuthorValidator)
```

File: pkg/handler/v1/view/book.go [hunk 2]
```diff
@@ -23,16 +29,33 @@ type BooksResponse = ListResponse[Book] // @name BooksResponse
...
+	field, ok := fl.Field().Interface().(string)
+	if ok {
+		// check if author is contain only letters and spaces
+		for _, char := range field {
+			if !(unicode.IsLetter(char) || unicode.IsSpace(char)) {
+				return false
+			}
+		}
+
+		return true
```
Comment (hieuphq) [inline pkg/handler/v1/view/book.go:57]: using regex

File: pkg/handler/v1/view/auth.go [hunk 3]
```diff
@@ -19,7 +25,65 @@ type Auth struct {
...
+		for _, char := range field {
+			if specialChars[char] {
+				containsSpecialChar = true
+				break
+			}
+		}
+
+		if !containsSpecialChar {
+			return false
+		}
```
Comment (hieuphq) [inline pkg/handler/v1/view/auth.go:83]: using regex

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/validator/author.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	field, ok := fl.Field().Interface().(string)
+	if ok {
+		pattern := "^[a-zA-Z ]+$"
+		regex := regexp.MustCompile(pattern)
+
+		return regex.MatchString(field)
+	}
+
+	return false
+}
```
Comment (hieuphq) [inline pkg/validator/author.go:20]: missing the unit test

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/validator/author.go (hunk 4 above)
Comment (datphamcode295) [reply pkg/validator/author.go:20]: updated

**Review by hieuphq — APPROVED** [review]

**Pull request merged**